com.unity.netcode.gameobjects/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -837,6 +837,58 @@ private void GetAllFieldsAndResolveGenerics(TypeDefinition type, ref List<TypeRe
             GetAllFieldsAndResolveGenerics(resolved, ref fieldTypes, genericParams);
         }
 
+        private void GetAllBaseTypesAndResolveGenerics(TypeDefinition type, ref List<TypeReference> baseTypes, Dictionary<string, TypeReference> genericParameters)
+        {
+
+            if (type.BaseType == null || type.BaseType.Name == "Object")
+            {
+                return;
+            }
+
+            var baseType = type.BaseType;
+
+            var genericParams = new Dictionary<string, TypeReference>();
+
+            if (baseType.IsGenericInstance)
+            {
+                var genericType = (GenericInstanceType)baseType;
+                var newGenericType = new GenericInstanceType(baseType.Resolve());
+                for (var i = 0; i < genericType.GenericArguments.Count; ++i)
```

Comment:
More a question than a comment on the code :). Is there a reason you're not using foreach apart from this is your preferred way to iterate? I'm just wondering if there are reasons not to use it which I'm not aware of?

Reply:
um... if you look at the code below, in the next loop, it's accessing two different arrays with the same index. I *think* when I originally wrote this loop, it was doing the same thing, so I was using C-style for to be able to have access to the index value for the other array. At some point I think I realized this loop doesn't need it and removed the other access, but didn't change the loop style.